src/components/EditableLabel/EditableLabel.tsx

```diff
@@ -0,0 +1,51 @@
+import { CSSProperties, FormEvent, KeyboardEvent, useRef, useState } from "react";
+import styled from "styled-components";
+
+interface RootBodyProps {
+  value?: string;
+  onBlur?: (value: string) => void;
+  className?: string;
+  style?: CSSProperties;
+}
+
+export interface Props extends RootBodyProps {}
+
+export default function EditableLabel({ value: initValue = "", onBlur = () => {} }: Props): JSX.Element {
+  const [value, setValue] = useState<string>(initValue);
+  const ref = useRef<HTMLInputElement>(null);
+
+  const handleChange = (e: FormEvent<HTMLInputElement>) => {
+    setValue(e.currentTarget.value);
+  };
+
+  const handleBlur = () => {
+    onBlur(value);
+  };
+
+  const handleKeyDown = ({ key }: KeyboardEvent) => {
+    if (key === "Enter" || key === "Escape") {
+      ref.current?.blur();
+
+      onBlur(value);
```

Comment:
[nitpick] - It looks like calling `handleBlur()` is more appropriate here. Not a big deal though. You can ignore this comment!